core/src/main/java/org/pentaho/di/connections/vfs/DefaultVFSConnectionManagerWrapper.java

```diff
@@ -0,0 +1,242 @@
+package org.pentaho.di.connections.vfs;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemOptions;
+import org.apache.commons.vfs2.provider.UriParser;
+import org.pentaho.di.connections.ConnectionDetails;
+import org.pentaho.di.connections.ConnectionManager;
+import org.pentaho.di.connections.vfs.provider.ConnectionFileName;
+import org.pentaho.di.connections.vfs.provider.ConnectionFileNameParser;
+import org.pentaho.di.core.bowl.Bowl;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleFileException;
+import org.pentaho.di.core.variables.VariableSpace;
+import org.pentaho.di.core.variables.Variables;
+import org.pentaho.metastore.api.exceptions.MetaStoreException;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+import static org.pentaho.di.connections.vfs.provider.ConnectionFileObject.DELIMITER_CHAR;
+
+public class DefaultVFSConnectionManagerWrapper extends BaseVFSConnectionManagerWrapper {
```

Comment:
`BaseVFSConnectionManagerWrapper` and `BaseVFSConnectionManagerWrapper` are split just as a way to separate complexity. The former is concerned with the more "manager" basic methods. The latter with the path-related and manager-test methods.